### PR TITLE
[Merged by Bors] - feat(Topology): lemas about `MapClusterPt`

### PR DESCRIPTION
**Pull request by urkud**
Also reuse `variable`s and make some arguments implicit.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Basic.lean [hunk 1]
```diff
@@ -940,31 +945,57 @@ theorem clusterPt_iff_ultrafilter {f : Filter X} : ClusterPt x f ↔
...
+    (hf : Tendsto f (𝓝 x ⊓ map u F) (𝓝 y)) (hu : MapClusterPt x F u) : MapClusterPt y F (f ∘ u) :=
+  (tendsto_inf.2 ⟨hf, tendsto_map.mono_left inf_le_right⟩).neBot (hx := hu)
+
+theorem Filter.Tendsto.comp_mapClusterPt [TopologicalSpace Y] {f : X → Y} {y : Y}
+    (hf : Tendsto f (𝓝 x) (𝓝 y)) (hu : MapClusterPt x F u) : MapClusterPt y F (f ∘ u) :=
+  (hf.mono_left inf_le_left).comp_mapClusterPt' hu
+
+theorem ContinuousAt.comp_mapClusterPt [TopologicalSpace Y] {f : X → Y} (hf : ContinuousAt f x)
+    (hu : MapClusterPt x F u) : MapClusterPt (f x) F (f ∘ u) :=
+  Tendsto.comp_mapClusterPt hf hu
```
Comment (YaelDillies) [inline Mathlib/Topology/Basic.lean:969]: I would expect `ContinuousAt.comp_mapClusterPt` to prove `ContinuousAt _ _`, not `MapClusterPt _ _ _`. What about
```suggestion
theorem MapClusterPt.tendsto_comp' [TopologicalSpace Y] {f : X → Y} {y : Y}
    (hf : Tendsto f (𝓝 x ⊓ map u F) (𝓝 y)) (hu : MapClusterPt x F u) : MapClusterPt y F (f ∘ u) :=
  (tendsto_inf.2 ⟨hf, tendsto_map.mono_left inf_le_right⟩).neBot (hx := hu)

theorem MapClusterPt.tendsto_comp [TopologicalSpace Y] {f : X → Y} {y : Y}
    (hf : Tendsto f (𝓝 x) (𝓝 y)) (hu : MapClusterPt x F u) : MapClusterPt y F (f ∘ u) :=
  (hf.mono_left inf_le_left).comp_mapClusterPt' hu

theorem MapClusterPt.continuousAt_comp [TopologicalSpace Y] {f : X → Y} (hf : ContinuousAt f x)
    (hu : MapClusterPt x F u) : (f x) F (f ∘ u) :=
  Tendsto.comp_mapClusterPt hf hu
```

**Review by YaelDillies — COMMENTED** [review]
The rest looks good

**Comment by YaelDillies** [comment]
maintainer merge

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Pull request closed without merging**